tensorboard/webapp/widgets/data_table/data_table_component.ts

```diff
@@ -144,6 +150,29 @@ export class DataTableComponent {
         return intlNumberFormatter.formatShort(
           selectedStepRunData.END_VALUE as number
         );
+      case ColumnHeaders.MIN_VALUE:
+        if (selectedStepRunData.MIN_VALUE === undefined) {
+          return '';
+        }
+        return intlNumberFormatter.formatShort(
+          selectedStepRunData.MIN_VALUE as number
```

Comment:
slightly odd that you have to cast this 'as number' given that you just checked whether it is undefined. What does the compiler think is the type of selectedStepRunData.MIN_VALUE?

(I guess comment applies multiple places in this switch statement)

Reply:
The type is number | string | undefined. So at this point the compiler thinks it is a number | string.